patchback/event_handlers.py

```diff
@@ -490,3 +520,31 @@ async def process_pr_backport_labels(
         text=f'Backported as {pr_resp["html_url"]}',
         summary=f'✅ {backport_pr_branch_msg!s}',
     )
+
+async def add_failure_label(
+    *,
+    labels_api: RepoLabelsAPI,
+    issue_labels_api: IssueLabelsAPI,
+    backport_label_prefix: str,
+    failed_label_prefix: str | None,
+    target_branch_prefix: str,
+    target_branch: str,
+) -> dict[str, Any] | None:
+    if failed_label_prefix is None:
+        return None
+    stripped_branch = target_branch[len(target_branch_prefix):]
+    label: str = failed_label_prefix + stripped_branch
+    backport_label = backport_label_prefix + stripped_branch
+    # Create label if it doesn't exist
+    try:
+        await labels_api.get_label(label)
+    except BadRequest as exc:
+        if exc.status_code != 404:
+            raise
+        await labels_api.create_label(
+            label, f'Failed to backport PR to {target_branch}.'
+        )
+    # Add failed label
+    await issue_labels_api.add_labels(label)
+    # Delete backport label
+    await issue_labels_api.remove_label(backport_label)
```

Comment:
I'm debating whether it makes sense to also drop the existing backport label on failures.

Reply:
A-la #12? That'd need to be a separate PR.